File: Nivedita/Array/Level2/Non-Repeating Element.cpp
class Solution{
    public:
    int firstNonRepeating(int arr[], int n) 
    { 
      map<int,pair<int,int>>mp;
       int j=INT_MAX;
       for(int i=0;i<n;i++)
       {
           mp[arr[i]]={i+1,mp[arr[i]].second+=1};
           
       }
       for(int i=0;i<n;i++)
       {
        if (mp[arr[i]].second==1)
        {
            return arr[i];
        }
           
       }
       return 0;
         // Complete the function
        
    } 
  
};

File: JyotiKumari/Array(GFG)/Non-Repeating element.py
from collections import OrderedDict

class Solution:
    def firstNonRepeating(self, arr, n): 
        # Complete the function
        answer = 0
        d = OrderedDict()
        for element in arr:
            d[element] = 0
        for element in arr:
            d[element]+= 1
        for key in d:
            if d[key]==1:
                answer = key
                break
        return answer